library/items/item_group/src/main/java/org/quiltmc/qsl/itemgroup/api/QuiltItemGroup.java

```diff
@@ -0,0 +1,192 @@
+	 *   		new Identifier("my_mod:item_group"))
+	 *   	.icon(() -> new ItemStack(MyMod.MY_ITEM))
+	 *   	.build();
+	 *
+	 *   public static Item MY_ITEM;
+	 *
+	 *   @Override
+	 *   public void onInitialize() {
+	 *     MY_ITEM = new Item(new QuiltItemSettings().group(MY_ITEM_GROUP));
+	 *   }
+	 * }
+	 * }</pre>
+	 */
+	public static final class Builder {
+		private final Identifier identifier;
+		private Supplier<ItemStack> stackSupplier = () -> ItemStack.EMPTY;
+		private Consumer<List<ItemStack>> stacksForDisplay;
+
+		/**
+		 * Create a new {@link Builder}. Using the constructor allows for the use of the {@link Builder#icon(Supplier)} and {@link Builder#appendItems(Consumer)} methods.
+		 * Manually setting the icon with {@link QuiltItemGroup#setIcon(ItemConvertible)} is possible after calling {@link Builder#build()}
+		 *
+		 * @param identifier the {@link Identifier} will become the name of the {@link ItemGroup} and will be used for the translation key
+		 */
+		public Builder(Identifier identifier) {
+			this.identifier = identifier;
+		}
+
+		/**
+		 * This is used to add an icon to the {@link ItemGroup}.
+		 *
+		 * @param stackSupplier the supplier should return the {@link ItemStack} that you wish to show on the tab
+		 * @return {@code this}
+		 */
+		public Builder icon(Supplier<ItemStack> stackSupplier) {
+			this.stackSupplier = stackSupplier;
+			return this;
+		}
+
+		/**
+		 * Adds a custom list of items to be displayed in a tab, such as items with enchantments, potions, or other NBT values.
+		 *
+		 * @param stacksForDisplay Add {@link ItemStack}s to this list to show in the {@link ItemGroup}
+		 * @return {@code this}
+		 */
+		public Builder appendItems(Consumer<List<ItemStack>> stacksForDisplay) {
+			this.stacksForDisplay = stacksForDisplay;
+			return this;
+		}
+
+		/**
+		 * This is a single method that makes creating an {@link ItemGroup} with an icon one call.
+		 *
+		 * @param identifier    the id will become the name of the {@link ItemGroup} and will be used for the translation key
+		 * @param stackSupplier the supplier should return the {@link ItemStack} that you wish to show on the tab
+		 * @return An instance of the created {@link ItemGroup}
+		 */
+		public static ItemGroup buildWithIcon(Identifier identifier, Supplier<ItemStack> stackSupplier) {
+			return new Builder(identifier).icon(stackSupplier).build();
+		}
+
+		/**
+		 * This is a single method that creates an {@link ItemGroup} one call. This method should only be used when setting the icon with {@link QuiltItemGroup#setIcon(ItemConvertible)}.
+		 *
+		 * @param identifier    the id will become the name of the {@link ItemGroup} and will be used for the translation key
+		 * @return An instance of the created {@link ItemGroup}
+		 */
+		public static QuiltItemGroup buildImmediate(Identifier identifier) {
+			return new Builder(identifier).build();
+		}
+
+		/**
+		 * @return An {@link ItemGroup} built from {@code this}
```

Comment:
```suggestion
		 * @return A new {@link ItemGroup}
```

built from `this` is implied